faust/_cython/streams.pyx

```diff
@@ -109,8 +109,7 @@ cdef class StreamIterator:
             object consumer
         consumer = self.consumer
         last_stream_to_ack = False
-        # if do_ack and event is not None:
-        if event is not None and (do_ack or event.value is self._skipped_value):
+        if do_ack and event is not None:
```

Comment:
I want to approve this, but I wish I could understand why `if do_ack and event is not None:` was disabled in https://github.com/faust-streaming/faust/pull/208. 

But, since you add `or value is skipped_value` below and pass this into this function, these changes are safe to merge.

Reply:
Everything is explained in https://github.com/faust-streaming/faust/pull/208#issuecomment-1488540746, thank you for that.

Reply:
I believe the intention was to ack the message anyway if it's skipped, even if `do_ack` is `False`, but this exact case was not properly tested. The mistake here is `event.value is self._skipped_value` will always return `False`. This is so because in this context `event.value` is the original value of the message, it can never be `skipped_value`.
Thanks for the approval!